app/Routes.php

```diff
@@ -406,7 +406,7 @@ function ( Request $request ) use ( $ffFactory ) {
 		$app->post(
 			'handle-paypal-payment-notification',
 			function ( Request $request ) use ( $ffFactory ) {
-				return ( new PayPalNotificationHandler( $ffFactory ) )->handle( $request );
+				return ( new PayPalNotificationController( $ffFactory ) )->handle( $request );
```

Comment:
You can clean these routes even more by making `FunFunFactory` a parameter of the `handle` methods of the controllers and giving the controller as a string here. See `AddDonationController` and `donation/add` in `Routes.php` for reference. The dependency injection framework will use the types of the `handle` parameters and automatically call it with the right instances in the right order. This will be a further step away from Silex towards Symfony.